tests/system.rs

```diff
@@ -145,6 +169,26 @@ fn print_mismatches(result: HashMap<String, Vec<Mismatch>>) {
     assert!(t.reset().unwrap());
 }
 
+pub fn run_rustfmt(filename: String, write_mode: WriteMode) -> String {
+    let sig_comments = read_significant_comments(&filename);
+    let mut config = get_config(sig_comments.get("config").map(|x| &(*x)[..]));
+
+    for (key, val) in &sig_comments {
+        if key != "target" && key != "config" {
+            config.override_value(key, val);
+        }
+    }
+
+    // Don't generate warnings for to-do items.
+    config.report_todo = ReportTactic::Never;
+
+    // Simulate run()
+    let mut out = Vec::new();
+    let file_map = format(Path::new(&filename), &config, write_mode);
+    let _ = filemap::write_all_files(&file_map, &mut out, write_mode, &config);
+    String::from_utf8(out).unwrap()
+}
```

Comment:
I think so. I'll can try to rework the idempotent test function to reuse this code if possible.